external/llvm-project/mlir/include/mlir/IR/AttrTypeBase.td

```diff
@@ -262,6 +262,60 @@ class AttrDef<Dialect dialect, string name, list<Trait> traits = [],
                                  "::" # cppClassName # ">($_self)">;
 }
 
+// Define an Attr name for NamedAttribute's.
+// - Name is dialect-scoped when not-inherent.
+// - Utilities to has/get/set/lookup on an Operation
+class AttrNameDef<Dialect dialect, string name, string userName,
```

Comment:
Maybe `DialectAttr`

Reply:
`AttrDef` already takes a Dialect.

Reply:
Good point, but AttrNameDef still doesn't quite feel right - I'd argue for something like `ExternalAttrDef`